File: src/game/BlockData.java
package game;

public class BlockData {
	
	private static enum BDState {ZERO, ONE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, UNKNOWN, FLAGGED};
	private final BDState state;
	
	public BlockData(boolean isClear, boolean isFlagged, byte nearbyMines) {
		if (isClear) {
			state = BDState.values()[nearbyMines];
		}
		else if (isFlagged) {
			state = BDState.FLAGGED;
		}
		else {
			state = BDState.UNKNOWN;
		}
	}

	public boolean isClear() {
		return state.ordinal() <= 8;
	}

	public boolean isFlagged() {
		return state == BDState.FLAGGED;
	}
	
	public boolean isUnknown() {
		return state == BDState.UNKNOWN;
	}

	public int getNearbyMines() {
		if (state.ordinal() > 8) {
			return -1;
		}
		else {
			return state.ordinal();
		}
	}
}

File: src/main/kotlin/game/BlockData.kt
package game

enum class BlockData {
	ZERO, ONE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, UNKNOWN, FLAGGED;

	val isClear: Boolean
		get() = ordinal <= 8

	val isFlagged: Boolean
		get() = this == FLAGGED

	val isUnknown: Boolean
		get() = this == UNKNOWN

	val nearbyMines: Int
		get() = if (ordinal > 8) {
			-1
		} else {
			ordinal
		}
}